src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -55,9 +62,56 @@ public Transaction createTransaction(Long fantasyTeamId, Long incomingPlayerId,
         }
 
         Player playerIn = playerService.getPlayerById(incomingPlayerId);
-        transaction.setPlayerIn(playerIn);
+        Player outgoingPlayer = playerService.getPlayerById(outgoingPlayerId);
+        String position = playerIn.getPosition();
+
+        if (playerIn.getTeam() == null) { //it means it is a free agent
```

Comment:
this condition should be if (transaction.getReceivingFantasyTeam() == null)
almost every real life player will have a team, because team represents their real life team, not their fantasy team